src/main/java/graphql/annotations/TypeFunction.java

```diff
@@ -17,10 +17,59 @@
 import graphql.schema.GraphQLType;
 
 import java.lang.reflect.AnnotatedType;
-import java.util.Collection;
 import java.util.function.BiFunction;
 
+/**
+ * A GraphQLType builder for java types.
+ */
 public interface TypeFunction extends BiFunction<Class<?>, AnnotatedType, GraphQLType> {
+    /**
+     * @deprecated Override {@link #buildType(String, Class, AnnotatedType)} as an alternative.
+     * This interface will not extend {@link BiFunction} in a future release.
+     */
+    @Deprecated
+    default GraphQLType apply(Class<?> aClass, AnnotatedType annotatedType) {
```

Comment:
renaming apply to buildType makes sense as a counterpart to canBuildType but does it need to be deprecated

Notice how much "disruption" you have caused in all the places in order to get a better name?

I would leave apply as a synonym and cause less disruption and live with the lesser name and simply document it as a synonym for the better named "buildType"

Either way this is a breaking change since Collection<Class<?>> getAcceptedTypes(); is SPI

If we are happy to break API then perhaps this disruption is ok